src/modm/driver/adc/ads101x.hpp

```diff
@@ -0,0 +1,282 @@
+            Input2 = int(ConfigRegister::MUX1),
+            Input3 = int(ConfigRegister::MUX1) | int(ConfigRegister::MUX0),
+            Input4 = int(ConfigRegister::MUX2),
+            Input5 = int(ConfigRegister::MUX2) | int(ConfigRegister::MUX0),
+            Input6 = int(ConfigRegister::MUX2) | int(ConfigRegister::MUX1),
+            Input7 = int(ConfigRegister::MUX2) | int(ConfigRegister::MUX1) | int(ConfigRegister::MUX0),
+        };
+        typedef Configuration<ConfigRegister_t, InputMultiplexer, (Bit14 | Bit13 | Bit12)> InputMultiplexer_t;
+
+        enum class
+        DataRate : uint16_t
+        {
+            Sps128 = 0,
+            Sps250 = int(ConfigRegister::DR0),
+            Sps490 = int(ConfigRegister::DR1),
+            Sps920 = int(ConfigRegister::DR1) | int(ConfigRegister::DR0),
+            Sps1600 = int(ConfigRegister::DR2),
+            Sps2400 = int(ConfigRegister::DR2) | int(ConfigRegister::DR0),
+            Sps3300 = int(ConfigRegister::DR2) | int(ConfigRegister::DR1)
+        };
+        typedef Configuration<ConfigRegister_t, DataRate, (Bit7 | Bit6 | Bit5)> DataRate_t;
+
+        enum class
+        ComparatorMode : uint16_t
+        {
+            Traditional = 0,
+            Window = int(ConfigRegister::COMP_MODE),
+        };
+        typedef Configuration<ConfigRegister_t, ComparatorMode, Bit4> ComparatorMode_t;
+
+        enum class
+        ComparatorPolarity : uint16_t
+        {
+            ActiveLow = 0,
+            ActiveHigh = int(ConfigRegister::COMP_POL),
+        };
+        typedef Configuration<ConfigRegister_t, ComparatorPolarity, Bit3> ComparatorPolarity_t;
+
+        enum class
+        ComparatorLatch : uint16_t
+        {
+            Nonlatching = 0,
+            Latching = int(ConfigRegister::COMP_LAT),
+        };
+        typedef Configuration<ConfigRegister_t, ComparatorLatch, Bit2> ComparatorLatch_t;
+
+        enum class
+        ComparatorQueue : uint16_t
+        {
+            // Enable the comparator function
+            // Assert the ALERT/RDY pin after the specified number of successive conversions exceeding the upper or lower threshold
+            OneConversion = 0,
+            TwoConversions = int(ConfigRegister::COMP_QUE0),
+            FourConversions = int(ConfigRegister::COMP_QUE1),
+
+            // Disable comparator queue and set the ALERT/RDY pin to a high-impedance state
+            Disable = int(ConfigRegister::COMP_QUE1) | int(ConfigRegister::COMP_QUE0)
+        };
+        typedef Configuration<ConfigRegister_t, ComparatorQueue, Bit4> ComparatorQueue_t;
+
+        struct modm_packed
+        Data
+        {
+            inline int16_t
+            getValue()
+            {
+                return static_cast<int16_t>((data[0] << 8) | data[1]) / 16;
+            }
+
+            inline float
+            getVoltage()
+            {
+                return getValue() * lsbSize;
+            }
+
+            uint8_t data[2];
+            float lsbSize;
```

Comment:
I would instead add a `uint8_t size` that is the index of the lsbSize array.
It adds a single indirection, but on Cortex-M it's practically the same speed.